recipe/meta.yaml

```diff
@@ -1,34 +1,90 @@
-{% set version = "2.2" %}
+{% set version = "2.2.1" %}
+{% set torchani_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: c086fe6cabe49a0408d1f1b12aaf14e33b9e6ef0ffb15f3020792934e7c9fbbe
-  
+  git_url: https://github.com/aiqm/{{ name }}.git
+  git_rev: {{ version }}
+
 build:
-  noarch: python
   number: 0
-  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win]
+  string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}                                                  # [cuda_compiler_version == "None"]
+  string: py{{ CONDA_PY }}cuda{{ cuda_compiler_version|replace('.', '') }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}  # [cuda_compiler_version != "None"]
+  script:
+    - "{{ PYTHON }} -m pip install . --global-option='--cuaev-all-sms' -vvv --no-deps"  # [cuda_compiler_version != "None"]
+    - "{{ PYTHON }} -m pip install . -vvv --no-deps"                                    # [cuda_compiler_version == "None"]
 
 requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
+    - sysroot_linux-64 ==2.17                # [linux64]
+
   host:
     - python
     - pip
     - setuptools_scm
+    - importlib_metadata
+    - numpy
+    - pytorch
+    - pytorch-gpu {{ pytorch_version }}  # [cuda_compiler_version != "None"]
+    - pytorch-cpu {{ pytorch_version }}  # [cuda_compiler_version == "None"]
+    # cuda 10.2 does not have cub, which need to be manually downloaded in setup.py
+    - wget                               # [cuda_compiler_version == "10.2"]
+    - unzip                              # [cuda_compiler_version == "10.2"]
+
   run:
     - python
-    - setuptools  # required for pkg_resources
-    - pytorch
+    - pip
+    - importlib_metadata
     - lark-parser
     - requests
+    - h5py
+    - numpy
+    - pytorch
+    # Notes: pin_compatible could be removed if we drop pytorch 1.8.0 support
+    # https://github.com/conda-forge/torchani-feedstock/pull/2#discussion_r739808104
+    - {{ pin_compatible('pytorch-gpu', max_pin='x.x.x') }}  # [cuda_compiler_version != "None"]
+    - {{ pin_compatible('pytorch-cpu', max_pin='x.x.x') }}  # [cuda_compiler_version == "None"]
 
 test:
   imports:
     - torchani
+    - torchani.cuaev                     # [cuda_compiler_version != "None"]
+  source_files:
+    - torchani/resources/*
+    - tests/*
+    - tools/*
+    - download.sh
+    - dataset/*
+    - test_requirements.txt
+  requires:
+    - wget
+    - unzip
+    - tensorboard
+    - tqdm
+    - ase
+    - h5py
+    - pillow
+    - pyyaml
+    - pytest
+    - expecttest
+  commands:
+    - python -m pip install pkbar                                             # [linux64]
```

Comment:
I think a recipe for pkbar should be made instead of installing it through pip here.

However, I think you both can decide if this is something yo uwant to do


```
 grayskull pypi pkbar
```

will help quite a bit.

Reply:
This is a fair point. See https://github.com/conda-forge/staged-recipes/pull/16698